accepted/0000-missing-command-lookups.md

```diff
@@ -0,0 +1,73 @@
+# Support Yarn style command/script/bin lookups from the CLI
+
+## Summary
+
+Allow for a CLI experience like:
+
+```sh
+npm install typescript
+npm tsc
+```
+
+## Motivation
+
+I have two main motivations:
+
+- Personal, looking at the list of yarn commands I [use in an average month](https://gist.github.com/orta/489dbfa1b7467804955ea7800adf1494#file-last_month_yarn-txt) (1200) roughly 30% (400 (^1)) of those commands are using the shorthand syntax to run a command installed into `node_modules/.bin` in a project.
+
+- Professional, it'd be great for the TypeScript docs to be able to simply say: `npm install typescript --save-dev' then `npm tsc index.ts` and there to be no ambiguity like our current recommendation for `npx tsc index.ts`.
+
+## Detailed Explanation
+
+When npm receives a command it does not recognise, npm first resolves the next string to see if it matches:
+
+ - Available "scripts" in `package.json`
+ - Commands matching in `./node_modules/.bin/`
+
+Failing that, then show help.
+
+## Rationale and Alternatives
+
+1. Having a `script`s section like this:
+
+    ```json
+    {
+      "scripts": {
+        "tsc": "tsc",
+        "eslint": "eslint",
+        "prettier": "prettier"
+      }
+    }
+    ```
+
+    Is a waste of space in the JSON, the `node_modules/.bin` folder is typically full of commands and after a while you either start prefixing commands with `node_modules/.bin` instead or you fill up the scripts section.
+
+2. This change removes the concept of "blessed" scripts. `npm test` works but `npm test:lint` does not. Eventually folks memorize the ones they can use shorthand for, but there's no reason for the distinction other than at some point it was decided to be given a short-cut. These special cases can be removed with this change.
+
+3. It does not break any existing setups. 
+ 
+ - `npm tsc` does not work today, anyone who is doing that is seeing a help message.
+ - `npm test:lint` also does not work today.
+
+
+Alternative to:
+
+- [RFC 279:for --default-command](https://github.com/npm/rfcs/pull/279) - tries to hit the same issue, but because it'd a user option it wouldn't be useful at the eco-system level and only people who know it exists would use it.
```

Comment:
The way I see it we can still have the option proposed in the `default-command` RFC. It would allow us to ship the ability to swap your `default-command` in npm7 and when npm8 comes we can simply turn the default value to `default-command=run-script`

Reply:
Yeah, that could definitely work if `run-script` was expanded to also include binaries 👍🏻

Reply:
I see, that would be a diff command then!

Reply:
It seems like pursuing a new command that was "`exec` without any option to install" and "fall back to `run-script`", and then seeing how useful that command is when set to the `default-command`, would be a viable and appropriately cautious path forward here?